xbmc/addons/AddonManager.cpp

```diff
@@ -815,6 +815,20 @@ void CAddonMgr::FindAddons(ADDON_INFO_LIST& addonmap, const std::string& path)
         AddonInfoPtr addonInfo = CAddonInfoBuilder::Generate(path);
         if (addonInfo)
         {
+          const auto& it = addonmap.find(addonInfo->ID());
+          if (it != addonmap.end())
+          {
+            if (it->second->Version() > addonInfo->Version())
+            {
+
+              CLog::Log(LOGWARNING, "CAddonMgr::{}: Addon '{}' already present on '{}' with higher version {}, other on '{}' with version {} becomes ignored",
+                          __FUNCTION__, addonInfo->ID(), it->second->Path(), it->second->Version().asString(), addonInfo->Path(), addonInfo->Version().asString());
+              continue;
+            }
+            CLog::Log(LOGDEBUG, "CAddonMgr::{}: Addon '{}' already present on '{}' with version {} becomes ignored and replaced with version {} on '{}'",
+                          __FUNCTION__, addonInfo->ID(), it->second->Path(), it->second->Version().asString(), addonInfo->Version().asString(), addonInfo->Path());
+          }
```

Comment:
Not sure where @ksooo's comment went, but I'd suggest the following:
```
               CLog::Log(LOGWARNING, "CAddonMgr::{}: Addon '{}' already present with higher version {} at '{}' - other version {} at '{}' will be ignored",
                           __FUNCTION__, addonInfo->ID(), it->second->Version().asString(), , it->second->Path(), addonInfo->Version().asString(), addonInfo->Path());
```
and
```
             CLog::Log(LOGDEBUG, "CAddonMgr::{}: Addon '{}' already present with version {} at '{}' will be replaced with version {} at '{}'",
                           __FUNCTION__, addonInfo->ID(), it->second->Version().asString(), it->second->Path(), addonInfo->Version().asString(), addonInfo->Path());
```

This consistently logs the version followed by path each time (I think I got the arguments correct!).

Reply:
> Not sure where @ksooo's comment went

I accidentally deleted my comment. Thanks for following up.

Reply:
Thanks is changed!